components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/api/impl/PolicyImpl.java

```diff
@@ -97,6 +102,22 @@ public Response addActivePolicy(PolicyWrapper policyWrapper) {
 
     private Response addPolicy(PolicyManagerService policyManagementService, ResponsePayload responseMsg,
                                org.wso2.carbon.policy.mgt.common.Policy policy) {
+        List<Device> devices = policy.getDevices();
+        if (devices != null && devices.size() == 1) {
```

Comment:
As a general practice we don't return null values from methods which returns a collection object. So  is it necessary to put a null check here?